infra/src/main/kotlin/fi/fta/geoviite/api/frameconverter/v1/FrameConverterServiceV1.kt

```diff
@@ -210,19 +244,14 @@ constructor(
                     locationTrackName = locationTrackNameOrNull,
                     locationTrackType = mappedLocationTrackTypeOrNull,
                 )
-
-            validRequest to emptyList()
-        } else {
-            val errorResponse = createErrorResponse(identifier = request.identifier, errors = nonNullErrors)
-
-            null to errorResponse
-        }
+            )
+        else Left(createErrorResponse(identifier = request.identifier, errors = nonNullErrors))
     }
 
     fun validateTrackAddressToCoordinateRequest(
         request: TrackAddressToCoordinateRequestV1,
         params: FrameConverterQueryParamsV1,
-    ): Pair<ValidTrackAddressToCoordinateRequestV1?, List<GeoJsonFeatureErrorResponseV1>> {
+    ): Either<List<GeoJsonFeatureErrorResponseV1>, ValidTrackAddressToCoordinateRequestV1> {
```

Comment:
🤔 Oon ainakin ite tottunut enemmän rakenteeseen, jossa virhe on oikealla, mutta eipä sinänsä kyllä juurikaan väliä. Mielekkäitä rakenteita nämä monadit ja arvovirheet ainakin meikäläisen mielestä

Reply:
Itse olen tottunut taas juuri toisin päin 😄 "If something is right, it belongs right". Ja joo, mielekkäitä rakenteita ovat, varsinkin tällaisiin käyttötapauksiin